### ISSUE TITLE
Assertion failure in 35_search_failure/Response format for search

### ISSUE THREAD

**Issue by jtibshirani** [elastic/elasticsearch#64367]
This seems related to #64012, which is a similar failure in the same test but with a different error. Perhaps this new error popped up after #64012 was fixed.

**Build scan**: https://gradle-enterprise.elastic.co/s/xycqj4zq7j3z6

**Repro line**: 
```
./gradlew ':modules:reindex:yamlRestTest' --tests "org.elasticsearch.index.reindex.ReindexClientYamlTestSuiteIT.test {yaml=reindex/35_search_failures/Response format for search failures}" \
  -Dtests.seed=B85FC33E2E252213 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-IQ \
  -Dtests.timezone=Antarctica/Vostok \
  -Druntime.java=14
```

**Reproduces locally?**: no

**Applicable branches**: master

**Failure history**: as far as I can tell, it's the first time we've seen this recently.

**Failure excerpt**:

```
 2> java.lang.AssertionError: Failure at [reindex/35_search_failures:32]: error.reason didn't match expected value:
                      error.reason: expected String [Partial shards failure] but was String [all shards failed]
        at __randomizedtesting.SeedInfo.seed([B85FC33E2E252213:300BFCE480D94FEB]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:414)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:391)
        at jdk.internal.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
```

**Comment by jtibshirani** [comment]
@dnhatn I was interested to learn more about this area so I reassigned this to me and opened #64382. I hope that's okay, looking forward to your suggestions on the PR.

**Comment by dnhatn** [comment]
Fixed in #64382
